src/go/k8s/api/redpanda/v1alpha2/common.go

```diff
@@ -57,6 +61,21 @@ func (s SASLMechanism) Equals(other SASLMechanism) bool {
 	return strings.EqualFold(string(s), string(other))
 }
 
+func (s *SASLMechanism) ScramToKafka() (kadm.ScramMechanism, error) {
+	if s == nil {
+		return 0, ErrUnsupportedSASLMechanism
+	}
+
+	switch {
+	case s.Equals(SASLMechanismScramSHA256):
```

Comment:
So, just FYI, the reason why we're doing something different here with case-insensitive matching is because I wanted to be compatible with both Strimzi (which uses all lowercase names, i.e. `scram-sha-256`) and our current mechanism specification (which uses all uppercase names, i.e. `SCRAM-SHA-256`).